atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderAppender.java

```diff
@@ -277,26 +276,47 @@ else if (member.getConfigTerm() < raft.getTerm()
     else if (member.getMember().getType() == RaftMember.Type.ACTIVE
         || member.getMember().getType() == RaftMember.Type.PROMOTABLE
         || member.getMember().getType() == RaftMember.Type.PASSIVE) {
-      final Snapshot snapshot = raft.getSnapshotStore().getCurrentSnapshot();
-      if (snapshot != null
-          && member.getSnapshotIndex() < snapshot.index()
-          && snapshot.index() >= member.getLogReader().getCurrentIndex()) {
-        if (!member.canInstall()) {
-          return;
-        }
-
-        log.debug("Replicating snapshot {} to {}", snapshot.index(), member.getMember().memberId());
-        sendInstallRequest(member, buildInstallRequest(member, snapshot));
-      } else if (member.canAppend()) {
-        sendAppendRequest(member, buildAppendRequest(member, -1));
-      }
+      replicateSnapshot(member);
     }
     // If no AppendRequest is already being sent, send an AppendRequest.
     else if (member.canAppend()) {
       sendAppendRequest(member, buildAppendRequest(member, -1));
     }
   }
 
+  private void replicateSnapshot(final RaftMemberContext member) {
```

Comment:
One thing to note that this is Atomix code and I haven't wrote that, if we talking about the general append stuff :sweat_smile: I just extracted the code and  tried to strech/pull out the conditional branches, because before I felt it was even worse. I wanted to extract it to a separate method to make clear that this part replicates the snapshot.  Maybe I rename it to try to replicate :sweat_smile: 

Not 100% sure what do you mean with the second point.

Reply:
Agreed on :

 * we use the old if statements
 * rename it to tryReplicateSnapshot